packages/amplify_datastore/lib/amplify_datastore_stream_controller.dart

```diff
@@ -0,0 +1,110 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+import 'package:amplify_datastore/types/DataStoreHubEvents/DataStoreHubEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/HubEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/ModelSyncedEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/NetworkStatusEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/OutboxMutationEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/SyncQueriesStartedEvent.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+import 'package:flutter/services.dart';
+
+import 'types/DataStoreHubEvents/OutboxStatusEvent.dart';
+
+EventChannel channel = const EventChannel("com.amazonaws.amplify/datastore_hub_events");
+ModelProviderInterface modelProvider;
+StreamSubscription eventStream;
+
+class DataStoreStreamController {
+  
+  StreamController get datastoreStreamController {
+    return controller;
+  }
+
+  void registerModelsForHub(ModelProviderInterface models) {
+    modelProvider = models;
+  }
+}
+
+StreamController controller = StreamController<DataStoreHubEvent>.broadcast(
+  onListen: onListen,
+  onCancel: onCancel,
+);
+
+onListen() {
+  if (eventStream == null ) {
+    eventStream = channel.receiveBroadcastStream(1).listen((event) {
+      switch(event["eventName"]) {
+        case "ready": {
+          controller.add(DataStoreHubEvent(event["eventName"]));
+        }
+        break;
+        case "networkStatus": {
+          controller.add(DataStoreHubEvent(event["eventName"], payload: NetworkStatusEvent(event)));
+        }
+        break;
+        case 'subscriptionsEstablished': {
+          controller.add(DataStoreHubEvent(event["eventName"]));
+        }
+        break;
+        case "syncQueriesStarted": {
+          controller.add(DataStoreHubEvent(event["eventName"], payload: SyncQueriesStartedEvent(event)));
+        }
+        break;
+        case "modelSynced": {
+          controller.add(DataStoreHubEvent(event["eventName"], payload: ModelSyncedEvent(event)));
+        }
+        break;
+        case "syncQueriesReady": {
+          controller.add(DataStoreHubEvent(event["eventName"]));
+        }
+        break;
+        case "outboxMutationEnqueued": {
+          try {
+            controller.add(DataStoreHubEvent(event["eventName"], payload: OutboxMutationEvent(event, modelProvider, event["eventName"])));
+          } catch (e) {
+            print(e);
+          }
```

Comment:
There's a lot of duplicated code inside each of the branches in this control flow. What can you do to shorten the implementation of each `case`?

Reply:
used a `rebroadcast` helper to at least shorten these lines a bit.